lib/ansible/parsing/vault/__init__.py

```diff
@@ -123,16 +135,25 @@ def is_encrypted(b_data):
 def is_encrypted_file(file_obj):
     """Test if the contents of a file obj are a vault encrypted data blob.
 
-    The data read from the file_obj is expected to be bytestrings (py2 'str' or
-    python3 'bytes'). This more or less expects 'utf-8' encoding.
-
     :arg file_obj: A file object that will be read from.
     :returns: True if the file is a vault file. Otherwise, False.
     """
     # read the header and reset the file stream to where it started
     current_position = file_obj.tell()
-    b_header_part = file_obj.read(len(b_HEADER))
-    file_obj.seek(current_position)
+    try:
+        file_obj.seek(0)
```

Comment:
It gives the caller the ability to optimize if they want but it will always be a tradeoff between accuracy and speed.  The caller could specify a count of 2048 for instance so that the header and a few bytes of the vaulttext are read.  That would be faster then reading the whole file.  But it could pass but when the rest of the vaulttext is later read, we could find out that there was non-ascii data after the first 2048 bytes and then decryption would fail.